src/buffer/out/Row.hpp

```diff
@@ -299,6 +305,8 @@ class ROW final
     bool _wrapForced = false;
     // Occurs when the user runs out of text to support a double byte character and we're forced to the next line
     bool _doubleBytePadded = false;
+
+    std::optional<ScrollbarData> _promptData = std::nullopt;

```

Comment:
@lhecker this will grow ROW by 12+ bytes (considering padding for alignment after `ScrollbarData::category`, and the size of the flag in an `optional`), for a total of 108k for a full buffer. Is this OK?

@zadjii-msft if you want to save the cost of this in conhost, you can use the preprocessor version of `TIL_FEATURE` for this and the implementation bodies (fwiw)

Reply:
Something else we discussed: 

```c++
struct ScrollbarData
{
    til::color color;
    uint32_t exitCode = 0;
    MarkCategory category{ MarkCategory::Default };

    bool valid = false;
    bool hasColor = false;
    bool hasExitCode = false;
};

struct ROW {
    // ...
    ScrollbarData _promptData;
};
```

to avoid some overhead

Reply:
While I'd agree that `std::optional` is optimal in keeping data uninitialized, I'd find it difficult to agree that it's optimal when it comes to its memory layout. `point`, `color` and so on for instance all double in size when wrapped in an `optional`.

The current `ScrollbarData` struct is 20 bytes large, whereas the one above is 12 bytes large. Due to ROW's necessary 16 byte alignment this effectively results in 20 bytes of padding per ROW.

`MarkExtents`' struct size for instance is 60 bytes, compared to just 28 if we didn't use `optional`. 54% of the struct is just empty space.